appveyor.yml

```diff
@@ -182,6 +194,239 @@ for:
     build_script:
       - "python -c \"import sys; print(sys.executable)\""
 
+    test_script:
+      # Pre-dev Tests
+      - "pip install -e \".[pre-dev]\""
+
+      # Runs only in Linux, logging Public ECR when running canary and cred is available
+      - sh: "
+        if [[ -n $BY_CANARY ]];
+          then echo Logging in Public ECR; aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws; 
+        fi"
+
+      - sh: "pytest -vv tests/integration/buildcmd"
+
+      # Set JAVA_HOME to java11
+      - sh: "JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64"
+      - sh: "pytest -vv tests/integration/buildcmd/test_build_cmd.py -k test_building_java11_in_process"
+
+  # Integ testing deploy, package, delete, and sync
+  -
+    matrix:
+      only:
+        - image: Ubuntu
+          configuration: DeployIntegTesting
+
+    install:
```

Comment:
These `install` steps are same for every testing group, can we keep them in a matrix where it will only applicable to `Ubuntu` image? And actual testing steps could be different for each testing group in a separate `matrix` definition.